### PR TITLE
feat(framework) Upload logs in `flwr-serverapp`

### PR DESCRIPTION
**Pull request by panh99**

### CODE REVIEW

File: src/py/flwr/common/constant.py [hunk 1]
```diff
@@ -87,6 +87,9 @@
 ISOLATION_MODE_SUBPROCESS = "subprocess"
 ISOLATION_MODE_PROCESS = "process"
 
+# Log streaming
+LOG_UPLOAD_INTERVAL = 0.2
```
Comment (danieljanes) [inline src/py/flwr/common/constant.py:91]: Do we want to make 5 requests per second? Every second might be sufficient, maybe even every other second.

Comment (panh99) [reply src/py/flwr/common/constant.py:91]: Hmmm, we won't really make 5 requests per second. If no new message, it won't make a request. It's actually the minimum interval for log uploading.

File: src/py/flwr/common/logger.py [hunk 2]
```diff
@@ -290,3 +298,58 @@ def restore_output() -> None:
...
+            pass
+
+        # Upload if any logs
+        if msgs:
+            req = PushLogsRequest(
+                node=node,
+                run_id=run_id,
+                logs=msgs,
+            )
+            stub.PushLogs(req)
```
Comment (danieljanes) [inline src/py/flwr/common/logger.py:331]: This needs error handling. We shouldn't crash the log uploader if a single request fails.

Comment (panh99) [reply src/py/flwr/common/logger.py:331]: Sure, that makes sense.

**Review by danieljanes — APPROVED** [review]

**Pull request merged**